src/_pytest/mark/structures.py

```diff
@@ -335,7 +332,7 @@ def markname(self) -> str:
         return self.name  # for backward-compat (2.4.1 had this attr)
 
     def __repr__(self) -> str:
-        return f"<MarkDecorator {self.mark!r}>"
+        return f"<MarkDecorator {repr(self.mark)}>"
```

Comment:
Any advice on how to unit test this, all that comes to mind is something like:

```
assert repr(MarkDecorator(mark="foobar", _ispytest=True)) == "MarkDecorator(mark='foobar')"
```

Im surprised codecov is complaining as I only changed the `!r` to call `repr` directly here, maybe its only on new diffs :)

Reply:
FWIW I don't see how this is better anyways :slightly_smiling_face:

Reply:
It looks to me like this __repr__ is not used and the one provided by I assume `@attr.s` (not familiar with that lib yet) is used?

Reply:
What I meant is that I don't see the benefit of `{repr(self.mark)}` over `{self.mark!r}`, given that `!r` is the usual way to get a repr in f-strings (and `str.format()`).

But you're right - attrs autogenerates a `__repr__` here, so this isn't needed at all. Looks like it was added when this class was originally written in 2009 (dcf194ebb8877da46af6fc97ce494e64dd1f766e), but then never removed when attrs was introduced in 2017 (07b2b18a01a3ba12757a4eb1c0e082922f5299e5).